### PR TITLE
Moved SetTestOptions to AdaptiveDialogBot

### LINKED ISSUE #5077: Officially support SetTestOptionsMiddleware
SetTestOptionsMiddleware is currently in Microsoft.Bot.Builder.Dialogs.Adaptive.Testing which is a preview package.  The next version of the emulator supports emitting the SetTestOptions event with randomValue/randomSeed to control randomness.  We need to make sure the SDK officially supports this middleware either by moving it into a release package like Microsoft.Bot.Builder.Dialogs.Adaptive or by removing preview on the testing package.  The testing package also includes support for mocking LUIS and for declarative adaptive dialog testing.

Additional notes:
TestOptions is currently being consumed in the adaptiveexpressions library ([here](https://github.com/Microsoft/botbuilder-dotnet/blob/3ea375eff52a5b57f4d8f7a02f374953a6534d32/libraries/AdaptiveExpressions/Extensions.cs#L67)).

### PR DESCRIPTION
**Pull request by tracyboehrer**
Fixes #5077

### CODE REVIEW

**Review by chrimc62 — APPROVED** [review]
:shipit:

**Comment by cleemullins** [comment]
@johnataylor Looking at this move, I'm not certain moving this Middleware from a preview pacakge into the Botbuilder package is what we want. Is there a way we could accomplish this w/o adding a new middleware component?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs [hunk 1]
```diff
@@ -125,6 +126,18 @@ private void SetUpTurnState(ITurnContext turnContext, BotFrameworkClient botFram
...
             turnContext.TurnState.Add(_telemetryClient);
 
+            // Catch "SetTestOptions" event and save into "Conversation.TestOptions".
+            if (turnContext.Activity.Type == ActivityTypes.Event)
+            {
+                var eventActivity = turnContext.Activity.AsEventActivity();
+                if (eventActivity.Name == "SetTestOptions")
+                {
+                    var conversationState = turnContext.TurnState.Get<ConversationState>();
+                    var property = conversationState.CreateProperty<object>("TestOptions");
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs:136]: @Stevenic, you said we use a different naming convention for "properties" what would be the right name here?

**Comment by tracyboehrer** [comment]
Closing this PR because nothing is as easy as it seems and now this is a mess.

**Pull request closed without merging**